crates/proof-of-sql-planner/src/util.rs

```diff
@@ -0,0 +1,243 @@
+use super::{PlannerError, PlannerResult};
+use arrow::datatypes::{Field, Schema};
+use datafusion::{
+    catalog::TableReference,
+    common::{Column, DFSchema, ScalarValue},
+};
+use proof_of_sql::base::database::{ColumnField, ColumnRef, ColumnType, LiteralValue, TableRef};
+
+/// Convert a [`TableReference`] to a [`TableRef`]
+///
+/// If catalog is provided it is ignored
+pub(crate) fn table_reference_to_table_ref(table: &TableReference) -> TableRef {
+    match table {
+        TableReference::Bare { table } => TableRef::from_names(None, table),
+        TableReference::Partial { schema, table } | TableReference::Full { schema, table, .. } => {
+            TableRef::from_names(Some(schema), table)
+        }
+    }
+}
+
+/// Convert a [`ScalarValue`] to a [`LiteralValue`]
+///
+/// TODO: add other types supported in `PoSQL`
+#[allow(dead_code)]
+pub(crate) fn scalar_value_to_literal_value(value: &ScalarValue) -> PlannerResult<LiteralValue> {
```

Comment:
Since this is always copied/cloned, let's take ownership of it. Users of the function can clone if needed.
```suggestion
pub(crate) fn scalar_value_to_literal_value(value: ScalarValue) -> PlannerResult<LiteralValue> {
```